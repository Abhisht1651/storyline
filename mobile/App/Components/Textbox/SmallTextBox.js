// @flow

import React, { useState, useEffect } from 'react'
import { View, Text, TextInput, KeyboardAvoidingView } from 'react-native'
import styles from '../Styles/TextboxStyles/SmallTextBoxStyle'
import Colors from '../../Themes/Colors'

type Props = {
  textColor: string,
  placeholder: string,
  placeholderColor: string,
  captureText: (args: string) => void
}

const SmallTextBox = (props: Props) => {
  const [value, setValue] = useState('')
  const { textColor, placeholder, placeholderColor, captureText } = props

  const textColorStyle = {
    color: textColor
  }

  useEffect(() => {
    captureText(value)
  }, [value])

  const changeTextHandler = (text: string) => {
    setValue(text)
  }
  return (
    <KeyboardAvoidingView behavior="padding">
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.headerText}>Write a bio</Text>
        </View>
        <View style={styles.innerContainer}>
          <TextInput
            placeholder={placeholder}
            multiline
            placeholderTextColor={placeholderColor}
            selectionColor={Colors.cursor}
            maxLength={75}
            spellCheck={false}
            autoCorrect={false}
            value={value}
            style={[styles.innerContainerText, textColorStyle]}
            onChangeText={changeTextHandler}
          />
        </View>
      </View>
    </KeyboardAvoidingView>
  )
}

SmallTextBox.defaultProps = {
  textColor: '#000080',
  placeholder: 'Up to 75 characters.',
  placeholderColor: '#8F8FAF'
}

export default SmallTextBox
