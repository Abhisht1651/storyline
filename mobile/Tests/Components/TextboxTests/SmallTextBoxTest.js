import { shallow } from 'enzyme'
import React from 'react'

import { TextInput } from 'react-native'
import SmallTextBox from '../../../App/Components/Textbox/SmallTextBox'

describe('SmallTextBox', () => {
  it('should match to snapshot', () => {
    const component = shallow(<SmallTextBox />)
    expect(component).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<SmallTextBox />)
    expect(wrapper.find('Text')).toHaveLength(1)
  })

  it('should change store when text is entered', () => {
    const component = shallow(<SmallTextBox />)
    component.find(TextInput).prop('onChangeText')('Green fox jumps over the wall')
    expect(component.find(TextInput).prop('value')).toEqual('Green fox jumps over the wall')
  })
})
