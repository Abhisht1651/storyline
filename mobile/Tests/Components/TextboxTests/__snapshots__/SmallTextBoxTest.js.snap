// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallTextBox should match to snapshot 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
>
  <View
    style={
      Object {
        "borderRadius": 10,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 29.57142857142857,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 24,
            "letterSpacing": -0.11,
            "margin": 0,
            "padding": 0,
          }
        }
      >
        Write a bio
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F5F8F8",
          "borderRadius": 10,
          "flex": 1,
          "marginTop": 19.714285714285715,
          "width": "100%",
        }
      }
    >
      <Component
        allowFontScaling={true}
        autoCorrect={false}
        maxLength={75}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Up to 75 characters."
        placeholderTextColor="#8F8FAF"
        rejectResponderTermination={true}
        selectionColor="#7B46E4"
        spellCheck={false}
        style={
          Array [
            Object {
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 21,
              "height": 134.71428571428572,
              "marginLeft": "5.43%",
              "marginTop": 32.857142857142854,
              "paddingTop": 0,
              "textAlignVertical": "top",
              "width": "94.20%",
            },
            Object {
              "color": "#000080",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</KeyboardAvoidingView>
`;
